crates/compiler/gen_dev/src/generic64/mod.rs

```diff
@@ -759,11 +763,63 @@ impl<
         }
     }
 
+    fn build_num_add_checked(
+        &mut self,
+        dst: &Symbol,
+        src1: &Symbol,
+        src2: &Symbol,
+        num_layout: &Layout<'a>,
+    ) {
+        use Builtin::Int;
+
+        let buf = &mut self.buf;
+
+        let fields = [*num_layout, Layout::bool()];
+        let return_layout = Layout::struct_no_name_order(&fields);
```

Comment:
Why is the layout passed in not the layout for the result type?

Reply:
changed that